### PR TITLE
core cleanup

### PR DESCRIPTION
**Pull request by dvrogozh**

### CODE REVIEW

File: _studio/mfx_lib/shared/src/libmfxsw_decode.cpp [hunk 1]
```diff
@@ -417,25 +413,34 @@ mfxStatus MFXVideoDECODE_DecodeHeader(mfxSession session, mfxBitstream *bs, mfxV
...
+
+    try
+    {
         // check existence of component
-        if (!session->m_pDECODE.get())
+        if (!session->m_pDECODE)
         {
             // create a new instance
             session->m_pDECODE.reset(session->Create<VideoDECODE>(*par));
             MFX_CHECK(session->m_pDECODE.get(), MFX_ERR_INVALID_VIDEO_PARAM);
```
Comment (mgonchar) [inline _studio/mfx_lib/shared/src/libmfxsw_decode.cpp]: .get() is unnecessary in such checks

File: _studio/shared/src/libmfx_core.cpp [hunk 2]
```diff
@@ -272,7 +254,7 @@ mfxStatus CommonCORE::LockFrame(mfxHDL mid, mfxFrameData *ptr)
             return MFX_ERR_INVALID_HANDLE;
         return (*pAlloc->Lock)(pAlloc->pthis, mid, ptr);
     }
-    catch(MFX_CORE_CATCH_TYPE)
```
Comment (mgonchar) [inline _studio/shared/src/libmfx_core.cpp]: MFX_CORE_CATCH_TYPE is defined as ```...```, so maybe we can get rid of the define as well?

File: _studio/mfx_lib/shared/src/libmfxsw_decode.cpp (hunk 1 above)
Comment (dvrogozh) [reply _studio/mfx_lib/shared/src/libmfxsw_decode.cpp]: It is unnesessary in the honest if (smart_ptr), but our MFX_CHECK transforms to if (!!!(smart_ptr)) and compiler goes crazy.

File: _studio/shared/src/libmfx_core.cpp (hunk 2 above)
Comment (dvrogozh) [reply _studio/shared/src/libmfx_core.cpp]: That's the last change in the series.

File: _studio/mfx_lib/shared/src/libmfxsw_decode.cpp (hunk 1 above)
Comment (mgonchar) [reply _studio/mfx_lib/shared/src/libmfxsw_decode.cpp]: got it

**Pull request merged**